async_processor/worker.py

```diff
@@ -198,72 +209,142 @@ async def _publish_response(
         )
 
 
+async def _iter_and_warn(maybe_generaotr):
+    if isinstance(maybe_generaotr, AsyncGeneratorType):
+        warnings.warn(
+            "Returning an AsyncGenerator from the AsyncProcessor.process method "
+            "is an experimental feature, and behaviours may change later.",
+            UserWarning,
+            stacklevel=1,
+        )
+        async for obj in maybe_generaotr:
+            yield obj
+        return
+    if isinstance(maybe_generaotr, GeneratorType):
+        warnings.warn(
+            "Returning an Generator from the Processor.process method "
+            "is an experimental feature, and behaviours may change later.",
+            UserWarning,
+            stacklevel=1,
+        )
+        for obj in maybe_generaotr:
+            yield obj
```

Comment:
Wouldn't this be blocking? We don't want to convert this to async iterator using threaded implementation?

Reply:
Also minor, typo
```suggestion
        for obj in maybe_generator:
            yield obj
```